lib/pbench/test/unit/server/test_user_create.py

```diff
@@ -54,74 +61,125 @@ def test_args(server_config, pytestconfig):
-        runner = CliRunner(mix_stderr=False)
-        result = runner.invoke(
-            user_create.main,
-            args=[
-                TestCreateUser.USER_SWITCH,
-                TestCreateUser.USER_TEXT,
-                TestCreateUser.EMAIL_SWITCH,
-                TestCreateUser.EMAIL_TEXT,
-                TestCreateUser.FIRST_NAME_SWITCH,
-                TestCreateUser.FIRST_NAME_TEXT,
-                TestCreateUser.LAST_NAME_SWITCH,
-                TestCreateUser.LAST_NAME_TEXT,
-                TestCreateUser.ROLE_SWITCH,
-                "ADMN",
-            ],
-            input=f"{TestCreateUser.PSWD_TEXT}\n",
-        )
-        assert result.exit_code == 2
+    def test_valid_users_list(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            TestUserManagement.test_args(server_config, pytestconfig)
+
+            result = runner.invoke(user_list,)
+            assert result.exit_code == 0
+            assert result.stdout == "{0:15}\t{1:15}\t{2:15}\t{3:15}\t{4:20}".format(
+                "Username", "First Name", "Last Name", "Registered On", "Email"
+            ) + "\n{0:15}\t{1:15}\t{2:15}\t{3:15}\t{4:20}\n".format(
+                "test_user", "First", "Last", "2021-04-16", "test@domain.com"
+            )
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_valid_users_update(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            TestUserManagement.test_args(server_config, pytestconfig)
+
+            result = runner.invoke(
+                user_update,
+                args=[
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    "new_test@domain.com",
+                ],
+            )
+            assert result.exit_code == 0
+            assert result.stdout == "User test_user updated\n"
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_admin_user_creation(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            result = runner.invoke(
+                user_create,
+                args=[
+                    TestUserManagement.USER_SWITCH,
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    TestUserManagement.EMAIL_TEXT,
+                    TestUserManagement.FIRST_NAME_SWITCH,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_SWITCH,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    TestUserManagement.ROLE_SWITCH,
+                    "ADMIN",
+                ],
+                input=f"{TestUserManagement.PSWD_TEXT}\n",
```

Comment:
While I understand that normal, security-minded humans are more likely to enter the password via the prompt, and, therefore, it is important to test that functionality, the test code itself is much simpler if you specify the password via the option value, so I recommend retaining one test case which tests the prompt and using the option for all the others.